pkg/cloudprovider/vsphere/instances.go

```diff
@@ -165,6 +166,11 @@ func (i *instances) InstanceExistsByProviderID(ctx context.Context, providerID s
 		return true, nil
 	}
 
+	if _, ok := os.LookupEnv("ENABLE_DELETION_OF_VM"); ok {
+		klog.V(4).Info("instances.InstanceExistsByProviderID() NOT FOUND with ", uid, ". Signaling deletion.")
+		return false, nil
```

Comment:
before returning `false, nil`  here, do we need to check the error type? `DiscoverNode` may have returned an err not related to the node not being found?

Reply:
Good call!